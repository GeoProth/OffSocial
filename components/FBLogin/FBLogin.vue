<template>
  <div class="fb-login">
    <div class="user-alert">
      <h1>Ready to get your life back?</h1>
      <p>Sometimes facebook will delete your cookies on accident</p>
      <p>Don't worry, this will only take a sec.</p>
    </div>
    <VFacebookLogin
      class="login-button"
      :app-id="$env.app_id"
      scope="name, email, id, picture, user_events, user_photos"
      version="v3.2"
      @login="loggedIn"
      @sdk-init="haveSdk"
    />
  </div>
</template>

<script>
import VFacebookLogin from "vue-facebook-login-component";
import throttle from "lodash/throttle";
export default {
  components: {
    VFacebookLogin
  },
  methods: {
    async loggedIn() {
      throttle(
        () =>
          window.FB.login(() => {}, {
            scope: "user_posts, user_photos, user_events, instagram_basic"
          }),
        10 * 1000
      );
      await this.$store.dispatch("facebook/setUserLoggedIn");
      this.$router.push("/dashboard");
    },
    async haveSdk({ FB }) {
      await this.$store.dispatch("facebook/getUserInfo");
    }
  }
};
</script>
<style lang="stylus" scoped>
.fb-login {
  display: grid;
  grid-template-columns: 1fr;
  align-items: center;
  text-align: center;
  p {
    margin-top: 0;
    margin-bottom: 0;
  }

  .user-alert {
    margin-bottom: 2rem;
  }
}
.login-button {
  width: 25%;
  margin: auto;
}
</style>
