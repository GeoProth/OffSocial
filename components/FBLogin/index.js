import FBLogin from "./FBLogin";
export default FBLogin;
