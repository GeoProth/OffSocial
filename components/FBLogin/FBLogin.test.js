import FBLogin from "./FBLogin";
describe("FBLogin", () => {
  let wrapper;
  const RouterPush = jest.fn();
  beforeEach(() => {
    wrapper = shallowMock(FBLogin, {
      mocks: {
        $router: {
          push: RouterPush
        },
        $env: {
          app_id: ""
        }
      }
    });
  });
  test("Renders", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
