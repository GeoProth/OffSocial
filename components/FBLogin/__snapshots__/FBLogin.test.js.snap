// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FBLogin Renders 1`] = `
<div class="fb-login">
  <div class="user-alert">
    <h1>Ready to get your life back?</h1>
    <p>Sometimes facebook will delete your cookies on accident</p>
    <p>Don't worry, this will only take a sec.</p>
  </div>
  <vfacebooklogin-stub value="[object Object]" appid="" version="v3.1" options="[object Object]" loginoptions="[object Object]" buttonstyle="[object Object]" loaderstyle="[object Object]" tokenstyle="[object Object]" textstyle="[object Object]" transition="" scope="name, email, id, picture, user_events, user_photos" class="login-button"></vfacebooklogin-stub>
</div>
`;
